### PR TITLE
Add fix for leaking client TEs

### PR DESCRIPTION
**Pull request by jchung01**
Adds/backports AppliedEnergistics/Applied-Energistics-2#5006. Not too sure if the issue(s) it fixed in the original mod are reproduceable in this fork, but better safe than sorry (especially since this code is [still used](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/master/src/main/java/appeng/blockentity/AEBaseBlockEntity.java#L393) in modern versions). I *think* this was causing `TickHandler` to hold onto references of old client objects on the client-side as a side effect, don't quote me on that though.

### CODE REVIEW

File: src/main/java/appeng/tile/AEBaseTile.java [hunk 1]
```diff
@@ -434,7 +434,16 @@ public void disableDrops() {
...
-        if (this.world != null) {
+        if (this.world == null) {
+            return;
+        }
+
+        // Clientside is marked immediately as dirty as there is no queue processing
+        // Serverside is only queued once per tick to avoid costly operations
+        // TODO: Evaluate if this is still necessary
+        if (this.world.isRemote) {
+            this.markDirty();
```
Comment (Rongmario) [inline src/main/java/appeng/tile/AEBaseTile.java:445]: Not necessary to mark chunks dirty on the client-side, this probably won't be a method that gets called from the client anyway.

**Review by Rongmario — APPROVED** [review]
Haven't read this portion of AE2's code before. `TickHandler` leaves a lot less to be desired, `World` instances should not be kept in maps like these, I can see it leaking in various scenarios. Perhaps using dimension IDs is a better way of keeping track of crafting jobs and whatnot.

This particular change looks to be fine however.

File: src/main/java/appeng/tile/AEBaseTile.java (hunk 1 above)
Comment (jchung01) [reply src/main/java/appeng/tile/AEBaseTile.java]: That's what I figured, I did just copy the existing code. Cleaned it up a bit in the following commit.

**Review by Rongmario — APPROVED** [review]

**Comment by Rongmario** [comment]
There is an issue with relevant code, but I'm going to talk over on discord and save it for another issue/PR. Nothing to do with client TEs.

**Pull request merged**